src/main/java/seedu/address/model/person/KeywordMatcherPredicate.java

```diff
@@ -0,0 +1,10 @@
+package seedu.address.model.person;
+
+import java.util.function.Predicate;
+
+/**
+ * Superclass for keyword matching predicates.
+ */
+public abstract class KeywordMatcherPredicate implements Predicate<Person> {
+
```

Comment:
Pull the toString method up here and the field keywords up to this class as well.

make a method that takes in a String stringToMatch and returns keywords.stream()
                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(stringToMatch, keyword));

Reply:
otherwise, making the abstract class would not be helpful. Its to reduce the repeated code and help with polymorphism.

Also there should be changes to any function taking in DescriptionContainsKeywordsPredicate, NameContainsKeywordsPredicate,... to take in varargs ContainsKeywordsPredicate.